Automattic/jetpack
# VideoPress: Fix video details form change detection

Fixes #29469
Fixes #29517
Fixes #29518

## Proposed changes:

* Moves the privacy setting to the formData from the video directly, allowing it to be changed like the other fields and fixing the initial state of the form when the video is not in the first page and is accessed directly
* Fixes change detection when changing the video poster by resetting the selected time and keeping the current media gallery image reference
* Fixes the loading UI when the video private settings are changed by returning the request function promise and adding it to the promises list

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the VideoPress dashboard with at least 7 videos (2 pages)
* Go to the details page of the video in the second page
* Check that the save button is initially disabled
* Refresh the page
* Check that the save button is initially disabled as well
* Make some changes and save
* Check that the save button is disabled after saving
* Change *only* the privacy setting
* Check that the page goes to a loading state while saving
* Change the video poster and save
* Check that the save button is disabled after save
